### DISCUSSION TITLE
Feature request: you can switch or toggle (Input Pin and Button) from 0 to 1 by keyboard shortcut

### DISCUSSION THREAD

**Discussion by Abod-Shashtari** [logisim-evolution/logisim-evolution#1382] [Ideas]
Its very useful to control the input pin and button by keyboard shortcut you assign like (NI Multisim) ,we can assign the shortcut key like the way we assign label so if I create 2 inputs pin (or Buttons) and the first one I give it "A" key as a shortcut and "B" key for the second input pin, I can control these 2 inputs by just pressing A and B Keys (from keyboard),as a result we can flip the pin value between 0 and 1 easily.

**Comment by maehne** [comment]
Nice idea! However, we must carefully check that this won't cause unintended side effects. For sure, there needs to be a dedicated setting, which would turn on/off this feature. Also, we typically use longer descriptive names for inputs, i.e., there would need to be a way to easily assign shortcuts to inputs. Let's see what other people think about this idea.

**Reply by primechaser** [reply]
agree, this would be a nice one, but tricky for any input pins larger than one bit. maybe to start with, such a feature could be applied to one bit sized input pins only (setting),

**Reply by Abod-Shashtari** [reply]
For input pins larger than one bit they can make it like counter ,So if the input pin is 2 bits by default its value is 00 but if keyboard shortcut pressed it will become 01 if the key pressed again then the value is 10 then 11 or another idea for 2 bits input pins will be 2 shortcut keys one for the first bit and the other for the second bit (and for 3 bits input pins will be 3 shortcut keys and so on ) , But yes its more useful on 1 bit input pins than larger input pins, So they can add this feature exclusively to 1 bit input pins.

**Comment by R3dst0ne** [comment]
I created #961 a few months ago.
Since they request the same feature, should we move the discussion there or continue here?

**Reply by maehne** [reply]
I agree that we should move further discussion to issue #961. For the moment, I referenced there this thread.
